File: src/main/java/de/twometer/amongus/gui/RoleRevealPage.java
package de.twometer.amongus.gui;

import de.twometer.amongus.model.PlayerRole;

public class RoleRevealPage extends BasePage {

    public RoleRevealPage() {
        super("RoleReveal.html");
        amongUs.getScheduler().runLater(4700, () -> amongUs.getGuiManager().showPage(new IngamePage()));
    }

    @Override
    public void onDomReady() {
        var players = amongUs.getSession().getPlayers();
        var me = amongUs.getSession().getMyself();
        context.call("setRole", me.role.name());

        for (var player : players) {
            if (me.role == PlayerRole.Impostor && player.role != me.role) continue;
            context.call("addTeammate", player.username);
        }
    }

}

File: src/main/kotlin/de/twometer/amogus/gui/RoleRevealPage.kt
package de.twometer.amogus.gui

import de.twometer.amogus.client.AmongUsClient
import de.twometer.neko.audio.SoundEngine

class RoleRevealPage : BasePage(path = "RoleReveal.html") {

    override fun onLoaded() {
        val me = AmongUsClient.session!!.myself
        call("setRole", me.role.name)
        AmongUsClient.session!!.players
            .filter { it.id != me.id }
            .forEach { call("addTeammate", it.username) }
        SoundEngine.play("RoleReveal.ogg")
        AmongUsClient.mainScheduler.runLater(4700L) {
            PageManager.overwrite(IngamePage())
        }
    }

}